clients/source.go

```diff
@@ -222,6 +226,10 @@ func (c *SourceClient) Terminate() error {
 			return err
 		}
 	}
+
+	// wait for log streaming to complete before returning
+	c.wg.Wait()
```

Comment:
this wont be called if `c.cmd.Process.Kill` fails. I think we need maybe to log + wrap all errors ?

Reply:
Oh good point; maybe we can do `defer c.wg.Wait()` at the top of the function?

Reply:
Interesting approach, should work

Reply:
but if `Kill` fails (and the plugin is still alive) we can't be sure it's going to close the channel/logs anyway? Also maybe log the `Kill` `err` before returning it, so that if that happens and there's a hang, the log is printed so we know what the cause may be?

Reply:
Good point... it seems fairly unlikely that killing the plugin would fail but that the plugin can still be alive and streaming logs at that point, so let's log it so we can identify if that happens, but otherwise leave it like this for now?

Reply:
Ok, updated to the way it was initially with an additional comment 👍

Reply:
Kill can't fail. it can only fail if the process already killed. I think we still need to do the

```
defer func() {
if err:= c.wg.wait(); err != nil {
   c.log().....
 }
}
```